### DISCUSSION TITLE
Const grouping should be removed

### DISCUSSION THREAD

**Discussion by RGBCube** [vlang/v#16568] [Ideas]
Reasons:
- Import grouping was removed before
- It is harder to search for consts
- 1 way: only one way should exist, now you can group them or don't, which is 2 ways

**Comment by spytheman** [comment]
I agree completely.

**Comment by Wajinn** [comment]
Respectfully disagree about this for a few reasons, and seek to present a different side of the argument:

1) DRY

	It can be annoying to have to keep retyping or to keep seeing the same keywords in almost succession.  If there are several consts in a row, not necessarily fun.

```
const a = 1
const b = 2
const c = 3
const d = 4
const e = 'more'
```

2) The const grouping is part of the original design, and includes being used in Go.

	To remove the grouping in V, arguably makes it look like a less evolved language, not the more evolved newer alternative that it is.

3) It's not really a violation of "only one way".

	A grouping is providing different functionality.  And if the argument is that it has to be one way or the other, then why not with` ()`, so that functionality is not being taken away.  It would be known to always search until the last `)`. Thus user could do as below:

```
const (countries = ["United States","Canada","United Kingdom", "Australia"])
// thus it can be the above or below
const (
	countries =  ["United States","Canada","United Kingdom", "Australia"]
	planets = ["Mars","Venus","Earth","Saturn"]
	fruits = ["Cherry","Apple","Peach","Pear"]
)
```

4) Can not ways be found that doesn't remove existing functionality and convenience?

	I can understand that everyone has their preference, but this is debatably a far reaching change, far into the development of the language.

**Comment by medvednikov** [comment]
Good reasons @Wajinn. I also prefer const grouping.

**Reply by Delta456** [reply]
Are there any plans to remove single line const then?

**Comment by JalonSolov** [comment]
You could use the same arguments for why `imports` _should_ be grouped...

**Comment by JoanaBLate** [comment]
I think that both sides have good reasons.

I remember the discussion about the unification of the "import" styles, when spytheman raised the winner reason: being able to "grep import something".

Manually searching where a const is declared inside a module is easier than manually searching all files that requires some module. Thus, the "grep" argument is not so important now.

In code, we see that importing uses a few lines, while declaring contants may take screens.

 I am OK if the "const" grouping becomes the only way.

Also, imho, the code looks prettier and more readable with dfferent styles: one for imports and other for constants.
